### PR TITLE
 gitignore patterns before globbing

### PR DESCRIPTION
**Pull request by janat08**
redo of  #118. I had autoformatting (linting) on, and hadn't realized it until too late. Still fixes both #97, and xojs/xo#235 since I had to refractor warnings for api.

### CODE REVIEW

**Comment by sindresorhus** [comment]
There are merge conflicts and ` Still fixes both #97, and xojs/xo#235 since` should still be separate PRs.

**Comment by janat08** [comment]
it's done

**Comment by janat08** [comment]
There're more conflicts, will you merge it this time?

**Comment by janat08** [comment]
Please merge and reward

**Comment by janat08** [comment]
@sindresorhus

**Comment by janat08** [comment]
@sindresorhus

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			"type": "node",
+			"request": "launch",
+			"name": "Run AVA test",
+			"program": "${workspaceFolder}/node_modules/ava/profile.js",
+			"args": ["${workspaceFolder}/test.js",
+				  "--serial"],
+			"skipFiles": ["<node_internals>/**/*.js"]
+		}
+	]
+}
```
Comment (sindresorhus) [inline .vscode/launch.json]: This should not be part of this PR. I'm not interested in editor-specific config here.

File: gitignore.js [hunk 2]
```diff
@@ -43,6 +43,12 @@ const reduceIgnore = files => {
 	}, gitIgnore());
 };
 
+const invseredIgnore = files => {
```
Comment (sindresorhus) [inline gitignore.js]: Typo

File: index.js [hunk 3]
```diff
@@ -22,13 +20,57 @@ const checkCwdOption = options => {
 	}
 };
 
+function gitPatterns(cb, sync = true) {
```
Comment (sindresorhus) [inline index.js]: You should not mix sync and async. Use two different methods for async and sync.

**Comment by sindresorhus** [comment]
This will need a lot of tests.

**Comment by janat08** [comment]
Dont know, 80$, half a year late, isn't really enough  for perfect implementation when fixing egregious code and perfect test coverage (from what I remember of your previous tests, I'm not sure what else is to be tested). If you reward me now, I can fix the conflicts, and run the tests again, so that I don't have to wait another half year. I actually specifically avoided your repos, because I accused you of bias, and then you compensated for that by demonstrating it.

**Comment by janat08** [comment]
?

**Comment by janat08** [comment]
?

**Comment by janat08** [comment]
Okay I'll resolve the code change requests, and the merge conflicts once again, if that's all that's needed to merge and reward. People are actually switching from globby to underlying library because of this issue.

**Comment by sindresorhus** [comment]
> Dont know, 80$

Issue bounties are not salary. They're meant as incentives for people to contribute.

> half a year late

I have a lot of PRs to review and yours requires a lot of review.

> I actually specifically avoided your repos, because I accused you of bias, and then you compensated for that by demonstrating it.

I have no idea what you're referring to, but I would indeed prefer it if you avoid my repos. I'd rather spend my time reviewing PRs where the submitter does a good job.

**Comment by sindresorhus** [comment]
I'm going to close this as it doesn't seem like you're going to resolve the inline comments or add tests like requested.
